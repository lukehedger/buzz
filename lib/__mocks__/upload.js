import { Buffer } from 'buffer'
import * as request from '../util/request'
import makeSwarmURL from '../util/makeSwarmURL'

const upload = (data, provider, schemeType = 'LONG') => {
  if (!provider) {
    throw new Error(
      'Swarm node provider not set. Use `buzz.setProvider` to set the Swarm provider.'
    )
  }

  const body = typeof data === 'string' ? Buffer.from(data) : data

  return request
    .post(body, makeSwarmURL(provider, 'raw', schemeType))
    .then(res => res.text())
}

export default upload
