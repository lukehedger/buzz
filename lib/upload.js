import { Buffer } from 'buffer'
import * as request from './util/request'
import makeSwarmURL from './util/makeSwarmURL'

/**
 * Upload data to Swarm and receive the resulting content hash
 *
 * @param  {Buffer|String} data                Data to upload
 * @param  {String}        provider            Swarm node provider
 * @param  {String}        [schemeType='LONG'] Swarm URI scheme type
 * @return {String}                            Swarm content hash of uploaded data
 */
const upload = (data, provider, schemeType = 'LONG') => {
  if (!provider) {
    throw new Error(
      'Swarm node provider not set. Use `buzz.setProvider` to set the Swarm provider.'
    )
  }

  const body = typeof data === 'string' ? Buffer.from(data) : data

  return request
    .post(body, makeSwarmURL(provider, 'raw', schemeType))
    .then(res => res.text())
}

export default upload
