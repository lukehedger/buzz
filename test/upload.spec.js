import upload from '../lib/upload'

jest.mock('../lib/upload')
jest.mock('../lib/util/request')

const SWARM_PROVIDER = 'http://localhost:8500'

describe('buzz :: upload', () => {
  it('Should upload content to Swarm', async () => {
    const actual = await upload('buzz', SWARM_PROVIDER)

    expect(actual).toMatchSnapshot()
  })

  it('Should throw an error if Swarm provider not set', () => {
    const actual = () => upload('buzz')

    expect(actual).toThrowErrorMatchingSnapshot()
  })
})
