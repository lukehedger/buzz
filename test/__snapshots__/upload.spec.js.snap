// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buzz :: upload Should throw an error if Swarm provider not set 1`] = `"Swarm node provider not set. Use \`buzz.setProvider\` to set the Swarm provider."`;

exports[`buzz :: upload Should upload content to Swarm 1`] = `"f070f4f87c0018a4026ff2c14895e9ef525e81c4cf4d768e1bf658e570927a13"`;
